src/js/controller/captions.js

```diff
@@ -66,13 +66,19 @@ const Captions = function(_model) {
             _tracksById = {};
             _unknownCount = 0;
         } else {
-            for (let i = 0; i < tracks.length; i++) {
-                _addTrack(tracks[i]);
-            }
-
-            // To avoid duplicate tracks in the menu when we reuse an _id, regenerate the tracks array
-            const allTracks = Object.keys(_tracksById).map(id => _tracksById[id]);
-            _tracks = allTracks.filter(track => !track.sideloaded).concat(allTracks.filter(track => !!track.sideloaded));
+            tracks.forEach(track => _addTrack(track));
+            // Make _tracks order match (video) tracks
+            _tracks.sort((trackA, trackB) => {
+                const videoTracksIndexOfA = tracks.indexOf(trackA);
+                const videoTracksIndexOfB = tracks.indexOf(trackB);
+                if (videoTracksIndexOfA === -1) {
```

Comment:
Just curious if it matters that there might be tracks like this that exist in `_tracks` but not in the video tracks? Could that cause anything?

Reply:
yeah that's what the indexOf check is for. When we come out of a midroll in Safari we are reloading the main video. Any in-band captions tracks temporarily are gone so we could have a side-loaded captions track (always first) in tracks, but here _tracks remembers all subtitles from before the ad. `if (videoTracksIndexOfA === -1) {` pushes items up or down to match the input.

Reply:
Tested via Chrome snippet:
```js
function align(a, b) {
    const before = a.join(', ');
    a.sort((_a, _b) => {
        const indexOfA = b.indexOf(_a);
        if (indexOfA === -1) {
            return 1;
        }
        const indexOfB = b.indexOf(_b);
        if (indexOfB === -1) {
            return -1;
        }
        return indexOfA - indexOfB;
    });
    console.log(before, a);
};

align(
    ['a', 'b', 'c'],
    ['a', 'b', 'c']
);

align(
    ['c', 'b'],
    ['a', 'b', 'c']
);

align(
    ['b', 'a'],
    ['a', 'b', 'c']
);

align(
    ['b', 'a', 'c'],
    ['a', 'b', 'c']
);

align(
    ['b', 'a', 'd', 'e'],
    ['a', 'b', 'c']
);

align(
    ['b', 'a', 'd', 'e', 'c'],
    ['a', 'b', 'c']
);
```
results (before > after)
```
a, b, c > ["a", "b", "c"]
c, b > ["b", "c"]
b, a > ["a", "b"]
b, a, c >["a", "b", "c"]
b, a, d, e > ["a", "b", "d", "e"]
```